### PR TITLE
Implement multithreading and Topics class

### LINKED ISSUE #213: Multithreading of SA fetch
Improve UX by now hanging UI when fetching from SA.

### LINKED ISSUE #214: Create Topic and Topics class to encapsulate weekly studio material

### PR DESCRIPTION
**Pull request by peironggg**
Fixes #213 
Fixes #214

### CODE REVIEW

File: src/main/java/seedu/address/model/topic/Topics.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.google.common.collect.ImmutableMap;
+
+public class Topics {
+    public final ImmutableMap<Integer, Topic> allTopics;
+
+    /**
+     * Initializes a new Topics object.
+     */
+    public Topics() {
```
Comment (Asuraxsoul) [inline src/main/java/seedu/address/model/topic/Topics.java:14]: good use of hashmap

File: src/main/java/seedu/address/scraper/ScraperManager.java [hunk 2]
```diff
@@ -96,19 +97,28 @@ public void startScraping() throws IOException {
             return;
         }
 
-        List<Mission> missions = getMissions();
-        List<Quest> quests = getQuests();
-        List<Student> students = new ArrayList<>();
+        // Run JavaFX-modifying functions after JavaFX Thread is done.
+        Platform.runLater(() -> {
```
Comment (Asuraxsoul) [inline src/main/java/seedu/address/scraper/ScraperManager.java:101]: Good idea to include threads for GUI!

**Review by Asuraxsoul — COMMENTED** [review]
Thanks for adding additional securities for GUI in general and the Topics field for GUI

File: src/main/java/seedu/address/model/topic/Topics.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        allTopics = ImmutableMap.copyOf(map);
+    }
+
+    /**
+     * Gets the topic for that week if it exists.
+     * @param week the week to be retrieved
+     * @return the Topic object that corresponds to what is covered that week
+     */
+    public Topic getTopic(int week) {
+        return allTopics.getOrDefault(week, new Topic());
```
Comment (wilinetan) [inline src/main/java/seedu/address/model/topic/Topics.java:36]: Nice use of this method!

File: src/main/java/seedu/address/scraper/ScraperManager.java (hunk 2 above)
Comment (wilinetan) [reply src/main/java/seedu/address/scraper/ScraperManager.java:101]: Agreed! Really helps to speed up the loading! :)

**Pull request merged**